_posts/2020-02-07-metal3-dev-env-install-deep-dive.md

```diff
@@ -0,0 +1,921 @@
+<br>
+
+> * **Images.sh.**  
+
+> The images.sh library file is sourced as well in the `01_prepare_host.sh` script . `Images.sh` contains multiple variables that sets the URL (`IMAGE_LOCATION`), name (`IMAGE_NAME`) and default username (`IMAGE_USERNAME`) of the cloud image that needs to be downloaded. The values of each variable will differ depending on the operating system of the virtual bare metal hosts. Note that these images will be served from the host to the virtual servers through the provisioning network.  
+
+> In our case, since `CentOS 7` is the base operating system, values will be defined as:
+<br>
+<br>
+
+> | **Name of the variable** | **Default value**      |
+> | IMAGE_NAME | CentOS-7-x86_64-GenericCloud-1907.qcow2 | 
+> | IMAGE_LOCATION | http://cloud.centos.org/centos/7/images | 
+> | IMAGE USERNAME | centos |
+
+<br>
+
+> **Information:** In case it is expected to use a custom cloud image, just modify the previous variables to match the right location.
+
+<br>
+Now that the cloud image is defined, the download process can be started. First, a folder defined by `IRONIC_IMAGE_DIR` should exist so that the image (`CentOS-7-x86_64-GenericCloud-1907.qcow2`) and its checksum can be stored. This folder and its content will be exposed through a local `ironic` container running in the host.
+
+<br>
+<br>
+
+| **Name of the variable** | **Default value**      |
+| IRONIC_IMAGE_DIR | /opt/metal3-dev-env/ironic/html/images |
+
+<br>
+Below it is verified that the cloud image files were downloaded successfully in the defined folder:
+
+```sh
+[alosadag@eko1 metal3-dev-env]$ ll /opt/metal3-dev-env/ironic/html/images
+total 920324
+-rw-rw-r--. 1 alosadag alosadag 942407680 Feb  3 12:39 CentOS-7-x86_64-GenericCloud-1907.qcow2
+-rw-rw-r--. 1 alosadag alosadag        33 Feb  3 12:39 CentOS-7-x86_64-GenericCloud-1907.qcow2.md5sum
+```
+
+Once the `images.sh` shared script is sourced, the following container images are pre-cached locally to the host in order to speed up things later. Below it is shown the code snippet in charge of that task:
+
+```sh
++ for IMAGE_VAR in IRONIC_IMAGE IPA_DOWNLOADER_IMAGE VBMC_IMAGE SUSHY_TOOLS_IMAGE DOCKER_REGISTRY_IMAGE
++ IMAGE=quay.io/metal3-io/ironic
++ sudo podman pull quay.io/metal3-io/ironic
+...
+....
+```
+The container image location of each one is defined by their respective variables: 
+<br>
+<br>
+
+| **Name of the variable** | **Default value**      |
+| VBMC_IMAGE | quay.io/metal3-io/vbmc |
+| SUSHY_TOOLS_IMAGE | quay.io/metal3-io/sushy-tools |
+| IPA_DOWNLOADER_IMAGE | quay.io/metal3-io/ironic-ipa-downloader	|
+| IRONIC_IMAGE | quay.io/metal3-io/ironic |
+| DOCKER_REGISTRY_IMAGE | docker.io/registry:latest |
+
+<br>
+
+> **Information:** In case it is expected to run self-modified container images, just overwrite the previous variables to match the right location. It is worth mentioning that a the container registry running as a privileged container in the host, so you can just upload there.
```

Comment:
It is worth mentioning that the container registry running as a privileged container in the host, so you can just upload there.

Reply:
> **Information:** In case it is expected to modify the public container images to test new features, it is worth mentioning that there is a container registry running as a privileged container in the host. Therefore it is recommended to upload your modified images there and just overwrite the previous variables to match the right location.